File: common/src/main/java/juuxel/adorn/criterion/AdornCriteria.java
package juuxel.adorn.criterion;

import juuxel.adorn.lib.registry.Registered;
import juuxel.adorn.lib.registry.Registrar;
import juuxel.adorn.lib.registry.RegistrarFactory;
import net.minecraft.advancements.triggers.CriterionTrigger;
import net.minecraft.core.registries.Registries;

public final class AdornCriteria {
    public static final Registrar<CriterionTrigger<?>> CRITERIA = RegistrarFactory.get().create(Registries.TRIGGER_TYPE);

    public static final Registered<SitOnBlockCriterion> SIT_ON_BLOCK =
        CRITERIA.register("sit_on_block", SitOnBlockCriterion::new);
    public static final Registered<BoughtFromTradingStationCriterion> BOUGHT_FROM_TRADING_STATION =
        CRITERIA.register("bought_from_trading_station", BoughtFromTradingStationCriterion::new);

    public static void init() {
    }
}

File: common/src/main/kotlin/juuxel/adorn/criterion/AdornCriteria.kt
package juuxel.adorn.criterion

import juuxel.adorn.lib.registry.Registrar
import juuxel.adorn.lib.registry.RegistrarFactory
import net.minecraft.advancement.criterion.Criterion
import net.minecraft.registry.RegistryKeys

object AdornCriteria {
    val CRITERIA: Registrar<Criterion<*>> = RegistrarFactory.get().create(RegistryKeys.CRITERION)
    val SIT_ON_BLOCK: SitOnBlockCriterion by CRITERIA.register("sit_on_block") { SitOnBlockCriterion() }
    val BOUGHT_FROM_TRADING_STATION: BoughtFromTradingStationCriterion by CRITERIA.register("bought_from_trading_station") {
        BoughtFromTradingStationCriterion()
    }

    fun init() {
    }
}
